packages/documents/lib/resolve.js

```diff
@@ -22,29 +22,34 @@ const getRepoId = (url, requireQuery) => {
   ])
 
   if (!url) {
-    return
+    return {}
   }
-  const {
-    hostname,
-    pathname,
-    searchParams
-  } = new URL(String(url), FRONTEND_BASE_URL)
+
+  const parsedUrl = new URL(String(url), FRONTEND_BASE_URL)
+  const { hostname, pathname, searchParams } = parsedUrl
+
   if (!pathname) { // empty for mailto
-    return
+    return { parsedUrl }
   }
+
   const pathSegments = pathname.split('/').filter(Boolean)
   if (
     hostname !== 'github.com' ||
     pathSegments.length !== 2 ||
     !GITHUB_ORGS.split(',').includes(pathSegments[0])
   ) {
-    return
+    return { parsedUrl }
   }
+
   if (requireQuery && !searchParams.has(requireQuery)) {
-    return
+    return { parsedUrl }
   }
+
+  searchParams.delete(requireQuery)
```

Comment:
~`searchParams` is not used again, why bother modifying it here?~ I see you use it on the returned `parsedUrl`. Could you add a comment explaining this. Also why does require query need to deleted on it anyway?

Reply:
`requireQuery` is either not set or set to `autoSlug` (thus far).

`autoSlug` in URL search params is more of a "flow" property ("do resolve this URL") and less one that should be propagated.

Using `searchParams.delete` is misleading. It "mutes "`parsedUrl.searchParams`. So using `parsedUrl.searchParams.delete` is more obvious.